pysdsl/test/test_rmq.py

```diff
@@ -0,0 +1,44 @@
+import random
+
+from pysdsl import IntVector, Int16Vector
+
+from pysdsl import rmq_sada, rmq_sct, rmq_sparse_tables
+
+
+
+def _test_rmq(rmq_class, container_class, target):
+    cont = container_class(50)
+    a = []
+
+    for i in range(50):
+        a.append(random.randint(0, 50))
+        cont[i] = a[i]
+
+    rmq = rmq_class(cont)
+
+    for _ in range(50):
+        l = random.randint(0, 49)
+        r = random.randint(l, 49)
+
+        i = rmq(l, r)
+
+        assert cont[i] == target(a[l:r+1])
+
+
+def test_rmq_sada():
+    _test_rmq(rmq_sada['Min'], IntVector, min)
```

Comment:
that can be parametrized, just provide a list of tuples to `pytest.mark.parametrized`. You also may want to have `names` argument in `parametrized`